subprojects/android-test/ui-testing-core-app/build.gradle.kts

```diff
@@ -93,76 +91,82 @@ instrumentation {
         fromSource.excludeFlaky = true
     }
 
-    val credentials = project.kubernetesCredentials
-    if (credentials !is Service && credentials !is KubernetesCredentials.Config) {
-        // todo fix this in MBS-11834
-        logger.warn("Instrumentation tasks are not created because kubernetes credentials not set")
-    } else {
-
-        afterEvaluate {
-            tasks.named("check").dependsOn(tasks.named("instrumentationUi"))
+    environments {
+        register<KubernetesViaCredentials>("kubernetes") {
+            token.set(getOptionalStringProperty("kubernetesToken"))
+            caCertData.set(getOptionalStringProperty("kubernetesCaCertData"))
+            url.set(getOptionalStringProperty("kubernetesUrl"))
+            namespace.set(getOptionalStringProperty("kubernetesNamespace"))
         }
+    }
 
-        val emulator22 = CloudEmulator(
-            name = "api22",
-            api = 22,
-            model = "Android_SDK_built_for_x86",
-            image = emulatorImage(22, "7bb4b0b720"),
-            cpuCoresRequest = "1",
-            cpuCoresLimit = "1.3",
-            memoryLimit = "4Gi"
-        )
+    experimental {
+        fetchLogcatForIncompleteTests.set(true)
+    }
 
-        val emulator29 = CloudEmulator(
-            name = "api29",
-            api = 29,
-            model = "Android_SDK_built_for_x86_64",
-            image = emulatorImage(29, "1927fb7cda"),
-            cpuCoresRequest = "1",
-            cpuCoresLimit = "1.3",
-            memoryLimit = "4Gi"
-        )
+    afterEvaluate {
```

Comment:
We should avoid `afterEvaluate` whenever possible. It gives unpredicted or implicit order for such blocks.

Can we wire tasks in convention plugins as we do in the main repo?  - "umbrella tasks" pattern.

Reply:
`afterEvaluate` isn't needed here. Replaced with 
```
tasks.check {
    dependsOn(tasks.named("instrumentationUiKubernetes"))
}
```

> Can we wire tasks in convention plugins as we do in the main repo? - "umbrella tasks" pattern.

Yes, we can.  But not in this PR, I think.